lib/random.gi

```diff
@@ -145,17 +143,22 @@ InstallMethod(Reset, [IsGAPRandomSource, IsObject], function(rs, seed)
   return old;
 end);
 
-InstallMethod(Random, [IsGAPRandomSource, IsList], function(rs, list)
-  local rx, rn;
-  if Length(list) < 2^28 then
+InstallMethod(Random, [IsGAPRandomSource, IsInt, IsInt], function(rs, l, u)
+  local gen; # rx, rn;
```

Comment:
Please, add a check of the arguments:
```
  if u < l then
    return fail;
  fi;
```

Reply:
Making this comment I notice that lines 213 and 215 also need a fix (it should be `d < 1` and `d = 1`, respectively.

Reply:
It would indeed be nice if such checks could be added, and perhaps also tests that verify them; if not in this PR, then in some later on.